game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/fire/battlegroup/NormalBattleGroups.java

```diff
@@ -0,0 +1,159 @@
+package games.strategy.triplea.delegate.battle.steps.fire.battlegroup;
+
+import static games.strategy.triplea.delegate.battle.BattleState.Side.DEFENSE;
+import static games.strategy.triplea.delegate.battle.BattleState.Side.OFFENSE;
+import static games.strategy.triplea.delegate.battle.BattleStepStrings.FIRST_STRIKE_UNITS;
+import static games.strategy.triplea.delegate.battle.BattleStepStrings.UNITS;
+
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.UnitType;
+import games.strategy.engine.data.properties.GameProperties;
+import games.strategy.triplea.Properties;
+import games.strategy.triplea.delegate.Matches;
+import games.strategy.triplea.delegate.battle.BattleState;
+import java.util.Collection;
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import lombok.experimental.UtilityClass;
+import org.triplea.java.PredicateBuilder;
+
+/**
+ * Creates battle groups for units that use attack/defense attributes.
+ *
+ * <p>There is a FirstStrike battle group and a General battle group.
+ */
+@UtilityClass
+class NormalBattleGroups {
+
+  /**
+   * Creates battle groups for the first strike phase
+   *
+   * <p>If an opposing isDestroyer unit is present, then the phase for that side will not occur (see
+   * {@link #noDestroyerPresent} for both firing squadrons)
+   */
+  static BattleGroup createFirstStrike(
+      final Collection<UnitType> unitTypes, final GameProperties properties) {
+
+    final List<FiringSquadron> firingSquadrons =
+        FiringSquadron.createWithTargetInformation(unitTypes).stream()
+            .map(
+                firingSquadron -> {
+                  return createFiringSquadron(
+                          firingSquadron, unitIsFirstStrike(properties), noDestroyerPresent())
+                      .name(FIRST_STRIKE_UNITS)
+                      .build();
+                })
+            .collect(Collectors.toList());
+
+    final BattleGroup.BattleGroupBuilder battleGroupBuilder =
+        BattleGroup.builder()
+            .offenseSquadrons(firingSquadrons)
```

Comment:
Could you help clarify why the same firingSquadrons can be both defense and offense?

Noting:
```
 .offenseSquadrons(firingSquadrons)
.defenseSquadrons(firingSquadrons)
```

Reply:
Okay, that helps. I was confused by the terminology, it is actually a configuration or a spec. Are we computing this per-battle, or once for any map?

It could be more clear to have a single variable to hold the valid sides. EG:
```
isValidFor(OFFENSE, DEFENSE)
```